libs/blocks/pdf-viewer/pdf-viewer.js

```diff
@@ -0,0 +1,50 @@
+import { getConfig } from '../../utils/utils.js';
+
+const API_SOURCE_URL = 'https://documentcloud.adobe.com/view-sdk/viewer.js';
+const PDF_RENDER_DIV_ID = 'adobe-dc-view';
+const {
+  env,
+  pdfViewerClientIdStage,
+  pdfViewerClientIdProd,
+  pdfViewerReportSuiteQA,
+  pdfViewerReportSuiteProd,
+} = getConfig();
+
+const init = (el) => {
+  const isProdEnv = env.name === 'prod';
+  const children = el?.querySelectorAll(':scope > div');
+  const loadApiScriptTag = document.createElement('script');
+  const url = children[0]?.querySelector('a[href$="pdf"]').href;
+  const fileName = url?.split('/').pop();
+
+  children[0]?.remove();
+
+  if (!url) return;
+
+  loadApiScriptTag.setAttribute('src', API_SOURCE_URL);
+  el?.appendChild(loadApiScriptTag);
+
+  const pdfViewerDiv = document.createElement('div');
+
+  pdfViewerDiv.setAttribute('id', PDF_RENDER_DIV_ID);
+  pdfViewerDiv.classList.add('pdf-container');
+  el?.appendChild(pdfViewerDiv);
```

Comment:
You can reuse the existing div so that you don't have to create/append a new one:

```suggestion
  const children = el?.querySelectorAll(':scope > div');
  if (!children?.length) return;

  const pdfViewerDiv = children[0];
  const url = pdfViewerDiv?.querySelector('a[href$="pdf"]').href;
  if (!url) return;

  pdfViewerDiv?.firstElementChild?.remove();
  pdfViewerDiv.setAttribute('id', PDF_RENDER_DIV_ID);
  pdfViewerDiv.classList.add('pdf-container');
  
  const loadApiScriptTag = document.createElement('script');
  loadApiScriptTag.setAttribute('src', API_SOURCE_URL);
  el?.appendChild(loadApiScriptTag);
  
  const fileName = url?.split('/').pop();
```